modules/afamqp/afamqp.c

```diff
@@ -645,7 +645,7 @@ afamqp_worker_insert(LogThreadedDestDriver *s, LogMessage *msg)
 {
   AMQPDestDriver *self = (AMQPDestDriver *)s;
 
-  if (!afamqp_dd_connect(self, TRUE))
+  if (!afamqp_dd_connect(self))
```

Comment:
`afamqp_dd_connect()` has a hidden connection check and disconnect/reconnect logic, which bypasses our `time-reopen()` functionality.

This `connect()` call could be removed from `insert()`, and the connection probe could be extracted from `insert()` as well if we decreased the default value of `time-reopen` at least for threaded destinations.

This is not the scope of this PR, so :+1: .